pkg/controller/operconfig/mtu_probe.go

```diff
@@ -0,0 +1,143 @@
+package operconfig
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/openshift/cluster-network-operator/pkg/apply"
+	"github.com/openshift/cluster-network-operator/pkg/render"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	cmNamespace = "openshift-network-operator"
+	cmName      = "mtu"
+)
+
+// probeMTU executes the MTU prober job, if the result configmap
+// doesn't exist. It then waits 100 seconds for results to be written,
+// then cleans up after itsef.
+// If, for whatever reason, it takes longer for the MTU to be detected,
+// it will adopt an existing job.
+func probeMTU(ctx context.Context, c crclient.Client) (int, error) {
+	mtu, err := readMTUConfigMap(ctx, c)
+	if err == nil {
+		_ = deleteMTUProber(ctx, c)
+		return mtu, nil
+	} else if !apierrors.IsNotFound(err) {
+		return 0, err
+	}
+
+	// cm doesn't exist, create Job
+	err = deployMTUProber(ctx, c)
+	if err != nil {
+		return 0, fmt.Errorf("failed to deploy mtu prober: %w", err)
+	}
+
+	log.Println("MTU prober deployed, waiting for result ConfigMap")
+
+	// wait up to 100 seconds for Job to report back.
+	tries := 10
```

Comment:
Can we use https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollImmediate instead of the for loop?

Reply:
yup